library/sha256.c

```diff
@@ -753,17 +769,29 @@ static const unsigned char sha256_test_sum[6][32] =
       0xF1, 0x80, 0x9A, 0x48, 0xA4, 0x97, 0x20, 0x0E,
       0x04, 0x6D, 0x39, 0xCC, 0xC7, 0x11, 0x2C, 0xD0 }
 };
+#endif
 
 /*
  * Checkup routine
  */
-int mbedtls_sha256_self_test( int verbose )
+static int mbedtls_sha256_common_self_test( int verbose, int is224 )
 {
-    int i, j, k, buflen, ret = 0;
+    int i, buflen, ret = 0;
     unsigned char *buf;
     unsigned char sha256sum[32];
     mbedtls_sha256_context ctx;
 
+    sha_test_sum_t* sha_test_sum;
+
+    sha_test_sum =
+#if defined(MBEDTLS_SHA224_C) && defined(MBEDTLS_SHA256_C)
```

Comment:
We're already checking the defines in `mbedtls_sha256_self_test` and `mbedtls_sha224_self_test`, no need to do that again here. We can just trust `is224`, it's a self_test, not anything that can have an input from outside.

Reply:
Actually this is most for symbols availability at C level. Depending on which between `MBEDTLS_SHA224_C` and `MBEDTLS_SHA256_C` is defined, then `sha224_test_sum` and `sha256_test_sum` might exist or not, therefore causing potential linking problems. Am I missing something here?

Reply:
Ah, right. I would suggest just writing the full line with `sha_test_sum = [,,]` in each case, it will be more readable this way :)